lib/madwf_tensor.cu

```diff
@@ -0,0 +1,123 @@
+#include <color_spinor_field.h>
+#include <tunable_nd.h>
+#include <device_vector.h>
+#include <madwf_transfer.h>
+#include <kernels/madwf_tensor.cuh>
+
+namespace quda
+{
+  namespace madwf_ml
+  {
+
+    template <class storage_type, class matrix_type> class tensor_5D_wrapper : public TunableKernel3D
```

Comment:
General style comment going forward.  For templated types, I think we should adopt the C++ convention of `storage_t` and `matrix_t`.